Pre-commit fails during pytest-check hook due to optional dependencies

### 🐛 Describe the bug
Optional dependencies are imported in tests run by pre-commit hook, so dev is required to run `poetry install --all-extras` before committing changes or pre-commit will fail.

### To Replicate
Install dependencies without extras using `poetry install`.
* Make sure pre-commit is installed: `pre-commit install`. Make changes and commit.
* Or simply run the tests. Specifically: `tests\llms\test_google_palm.py`

### Error
```
tests/llms/test_google_palm.py:None (tests/llms/test_google_palm.py)
ImportError while importing test module 'C:\Users\Jonathan\OneDrive\Documents\GitHub\pandas-ai\tests\llms\test_google_palm.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
..\.venv\lib\site-packages\_pytest\python.py:617: in _importtestmodule
    mod = import_path(self.path, mode=importmode, root=self.config.rootpath)
..\.venv\lib\site-packages\_pytest\pathlib.py:564: in import_path
    importlib.import_module(module_name)
..\..\..\..\..\AppData\Local\Programs\Python\Python39\lib\importlib\__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
<frozen importlib._bootstrap>:1030: in _gcd_import
    ???
<frozen importlib._bootstrap>:1007: in _find_and_load
    ???
<frozen importlib._bootstrap>:986: in _find_and_load_unlocked
    ???
<frozen importlib._bootstrap>:680: in _load_unlocked
    ???
..\.venv\lib\site-packages\_pytest\assertion\rewrite.py:172: in exec_module
    exec(co, module.__dict__)
llms\test_google_palm.py:5: in <module>
    from google import generativeai
E   ImportError: cannot import name 'generativeai' from 'google' (unknown location)
```

### Proposed Solution
* Add `poetry install --all-extras` to CONTRIBUTING.md. I do think this adds a bit of hassle for development and foresee this issue coming up in the future.
* @mspronesti mentions a `pytest.importskip` in #171, would that also work locally?

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -23,7 +23,7 @@ We use `poetry` as our package manager. You can install poetry by following the 
 Please DO NOT use pip or conda to install the dependencies. Instead, use poetry:
 
 ```bash
-poetry install
+poetry install --all-extras
 ```
 
 ### 📌 Pre-commit

